esmvaltool/preprocessor/_area_pp.py

```diff
@@ -108,22 +123,71 @@ def area_average(cube, coord1, coord2):
         cube: iris.cube.Cube
             input cube.
 
-        coord1: str
-            name of first coordinate
-
-        coord2: str
-            name of second coordinate
+        coord1, coord2: str, str
+            coords to use
 
     Returns
     -------
     iris.cube.Cube
         collapsed cube.
     """
     # check for bounds just in case
-    for coord in (coord1, coord2):
-        if not cube.coord(coord).has_bounds():
-            cube.coord(coord).guess_bounds()
+    coords = [coord1, coord2]
+    cube = _guess_bounds(cube, coords)
     grid_areas = iris.analysis.cartography.area_weights(cube)
-    result = cube.collapsed(
-        [coord1, coord2], iris.analysis.MEAN, weights=grid_areas)
+    result = cube.collapsed(coords, iris.analysis.MEAN, weights=grid_areas)
     return result
+
+
+def _area_average_general_cecks(coords, aggregator, weighted=False):
+    """Perform a set of pre-checks before area_average_general."""
+    if coords is None:
+        coords = ['latitude', 'longitude']
+
+    # Make sure that aggregator is an Aggregator instance
+    if not isinstance(aggregator, iris.analysis.Aggregator):
+        raise ValueError('Aggregator not an iris aggregator!')
+    # If doing weighted aggregation make sure that aggregator
+    # is a WeightAggregator instance
+    if weighted:
+        if not isinstance(aggregator, iris.analysis.WeightedAggregator):
+            raise ValueError('Aggregator not a weighted iris aggregator!')
+
+    return coords, aggregator
+
+
+def area_average_general(cube,
```

Comment:
This function is almost the same as `zonal_means` and `area_average`/`average_region`, I do not think we need yet another function that does the same thing. I would prefer to have one function, for example called `area_statistics`,  that can do all of these tasks, i.e. perform a statistic function or list of statistics functions from iris.analysis (with the name specified in the recipe, defaulting to mean) over a list of coordinates that is specified in the recipe, optionally using weights.